src/datadog/tracer_config.cpp

```diff
@@ -119,6 +122,134 @@ Expected<std::unordered_map<std::string, std::string>> parse_tags(
+  const auto si = env::DD_PROPAGATION_STYLE_INJECT;
+  // clang-format off
+  /*
+           ts    tse   se    tsi   si
+           ---   ---   ---   ---   ---
+    ts  |  x     warn  warn  warn  warn
+        |
+    tse |  x     x     warn  ok    ok
+        |
+    se  |  x     x     x     ok    ok
+        |
+    tsi |  x     x     x     x     warn
+        |
+    si  |  x     x     x     x     x
+  */
+  // In each pair, the first would be overridden by the second.
+  const std::pair<env::Variable, env::Variable> questionable_combinations[] = {
+           {ts, tse}, {ts, se},  {ts, tsi}, {ts, si},
+
+                      {se, tse}, /* ok */   /* ok */
+
+                                 /* ok */   /* ok */
+
+                                            {si, tsi},
+  };
+  // clang-format on
+
+  const auto warn_message = [](StringView name, StringView value,
+                               StringView name_override,
+                               StringView value_override) {
+    std::string message;
+    message += "Both the environment variables ";
+    append(message, name);
+    message += "=";
+    message += json_quoted(value);
+    message += " and ";
+    append(message, name_override);
+    message += "=";
+    message += json_quoted(value_override);
+    message += " are defined. ";
+    append(message, name_override);
+    message += " will take precedence.";
+    return message;
+  };
+
+  for (const auto &[var, var_override] : questionable_combinations) {
+    const auto value = lookup(var);
+    if (!value) {
+      continue;
+    }
+    const auto value_override = lookup(var_override);
+    if (!value_override) {
+      continue;
+    }
+    const auto var_name = name(var);
+    const auto var_name_override = name(var_override);
+    logger.log_error(Error{
+        Error::MULTIPLE_PROPAGATION_STYLE_ENVIRONMENT_VARIABLES,
+        warn_message(var_name, *value, var_name_override, *value_override)});
+  }
+
+  // Parse the propagation styles from the configuration and/or from the
+  // environment.
+  // Exceptions make this section simpler.
+  try {
+    const auto global_styles = styles_from_env(env::DD_TRACE_PROPAGATION_STYLE);
+    result.extraction_styles =
+        value_or(styles_from_env(env::DD_TRACE_PROPAGATION_STYLE_EXTRACT),
+                 styles_from_env(env::DD_PROPAGATION_STYLE_EXTRACT),
+                 global_styles, config.extraction_styles);
+    result.injection_styles =
+        value_or(styles_from_env(env::DD_TRACE_PROPAGATION_STYLE_INJECT),
+                 styles_from_env(env::DD_PROPAGATION_STYLE_INJECT),
+                 global_styles, config.injection_styles);
+  } catch (Error &error) {
+    return std::move(error);
+  }
+
+  if (!result.extraction_styles.datadog && !result.extraction_styles.b3 &&
+      !result.extraction_styles.none) {
+    return Error{Error::MISSING_SPAN_EXTRACTION_STYLE,
+                 "At least one extraction style must be specified."};
```

Comment:
Could we just fall back to a default in this instance, ie: datadog extraction / injection.

Reply:
True. The thing is, the only way to get this error is if the user explicitly said "the list of styles is empty."

If empty is a suitable default, then why have an explicit "none" style, especially if "none" only makes sense by itself? Better to just have "no styles" mean "no propagation."

Alas, "none" is in our spec.

Since the easiest way to get the default behavior is to not configure the styles, I'd rather have this error than add another way to get the default behavior.